distributed/diagnostics/cudf.py

```diff
@@ -0,0 +1,25 @@
+"""
+Diagnostics for memory spilling managed by cuDF.
+"""
+
+from __future__ import annotations
+
+try:
+    from cudf.core.buffer.spill_manager import get_global_manager
+except ImportError:
+    get_global_manager = None
+
+
+def real_time():
+    if get_global_manager is None:
+        return {"cudf-spilled": 0}
+    mgr = get_global_manager()
+    if mgr is None:
+        return {"cudf-spilled": 0}
+
+    totals = mgr.statistics.spill_totals
+
+    return {
+        "cudf-spilled": totals.get(("gpu", "cpu"), (0,))[0]
+        - totals.get(("cpu", "gpu"), (0,))[0]
```

Comment:
I'm making an assumption here that unspilled memory is freed from host and can be excluded from the total number of spilled bytes - is this a fair assumption to make? If not, will probably need to calculate the nbytes currently on host in a different way

Reply:
That's a good question, it seems like a sensible assumption but perhaps would be better if @madsbk or @wence- could confirm, they should know for sure what happens.

Reply:
Every time a spilling event occurs, the relevant pair is incremented with the number of bytes spilled. So for any given pair, the entry is always monotonic increasing.

If you want to track the amount of currently spilled memory, then yes, you want to know "how much was spilled from gpu to cpu" and "how much was spilled from cpu to gpu", and the difference tells you how much is _currently_ spilled to cpu memory.